group-coordinator/src/main/java/org/apache/kafka/coordinator/group/util/SystemTimerReaper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.util;
+
+import org.apache.kafka.server.util.ShutdownableThread;
+import org.apache.kafka.server.util.timer.Timer;
+import org.apache.kafka.server.util.timer.TimerTask;
+
+/**
+ * SystemTimerReaper wraps a {@link Timer} and starts a reaper thread
+ * to expire the tasks in the {@link Timer}.
+ */
+public class SystemTimerReaper implements Timer {
+    private static final long WORK_TIMEOUT_MS = 200L;
+
+    class Reaper extends ShutdownableThread {
+        Reaper(String name) {
+            super(name, false);
+        }
+
+        @Override
+        public void doWork() {
+            try {
+                timer.advanceClock(WORK_TIMEOUT_MS);
```

Comment:
How did we decide upon 200? Is my understanding correct that we only execute the event if it expiration is within 200 ms of the current time?

Reply:
To be honest, I don't know. I just reused the magic number that we already used:
* https://github.com/apache/kafka/blob/8d24716f27b307da79a819487aefb8dec79b4ca8/core/src/main/scala/kafka/raft/TimingWheelExpirationService.scala#L56
* https://github.com/apache/kafka/blob/8d24716f27b307da79a819487aefb8dec79b4ca8/core/src/main/scala/kafka/server/DelayedOperation.scala#L444